doc/source/projections.rst

```diff
@@ -98,6 +98,41 @@ Re-projecting is the process of changing the representation of locations from on
     ax.set_title("Mercator");
 
 
+Projection for multiple geometry columns
+----------------------------------------
+
+GeoPandas 0.8 implements the support for different projections assigned to different geometry
+columns of the same GeoDataFrame. Projection is now stored together with geometries directly
+on the GeometryArray level.
+
+Note that if GeometryArray has assigned projection, it is preferred over the
+projection passed to GeoSeries or GeoDataFrame during the creation:
```

Comment:
Reading this here, I am still not really convinced this is the best option ..

I know it is like this for GeoDataFrame right now, but not sure we should repeat the same "mistake" for arrays.

Reply:
But now reading the code below, should the example here not show a warning? (as the user would see if there is a CRS mismatch)